code/game/objects/items/tools/surgery_tools.dm

```diff
@@ -195,16 +195,55 @@
  */
 
 /obj/item/tool/surgery/bonegel
-	name = "bone gel"
+	name = "bottle of bone gel"
+	desc = "A container for bone gel that often needs to be refilled from a specialized machine."
+	desc_lore = "Bone gel is a biological synthetic bone-analogue with the consistency of clay. It is capable of fixing hairline fractures and complex fractures alike. Bone gel should not be used to fix missing bone, as it does not replace the body's bone marrow. Overuse in a short period may cause acute immunodeficiency or anemia."
 	icon_state = "bone-gel"
-	force = 0
-	throwforce = 1
 	w_class = SIZE_SMALL
 	matter = list("plastic" = 7500)
+	///percent of gel remaining in container
+	var/remaining_gel = 100
+	///If gel is used when doing bone surgery
+	var/unlimited_gel = FALSE
+	///Time it takes per 10% of gel refilled
+	var/time_per_refill = 2.5 SECONDS
+
+/obj/item/tool/surgery/bonegel/get_examine_text(mob/user)
+	. = ..()
+	if(!unlimited_gel) //Only show how much gel is left if it actually uses bone gel
+		. += "A counter on the side reads \"[remaining_gel]% gel remaining\"."
+
+/obj/item/tool/surgery/bonegel/proc/refill_gel(obj/refilling_obj, mob/user)
+	if(unlimited_gel)
+		to_chat(user, SPAN_NOTICE(" [refilling_obj] refuses to fill [src]."))
+		return
+	if(remaining_gel >= 100)
+		to_chat(user, SPAN_NOTICE("[src] cannot be filled with any more bone gel."))
+		return
+
+	while(remaining_gel < 100)
+		if(!do_after(user, time_per_refill, INTERRUPT_ALL, BUSY_ICON_MEDICAL, refilling_obj))
+			continue
```

Comment:
Isn't that meant to be break here rather than continue? You'd just keep on trying to refill forever if you move away